### ISSUE TITLE
About encode method of FixedPointEncoder for numpy array

### ISSUE THREAD

**Issue by matsutakk** [facebookresearch/CrypTen#484]
Hi, thank you for the great MPC framework.

I think we need parentheses here.

https://github.com/facebookresearch/CrypTen/blob/6ef151101668591bcfb2bbf7e7ebd39ab6db0413/crypten/encoder.py#L62
should be
```python
 return (self._scale * torch.from_numpy(x)).long().to(device)
```

Thank you.

**Comment by matsutakk** [comment]
I noticed that when executing below code, decreasing accuracy.


```python

from sklearn.datasets import load_iris,load_breast_cancer,load_digits
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import StratifiedKFold
from sklearn.metrics import accuracy_score
import numpy as np

data = load_iris()
X = data.data
y = data.target

k = 1
knn = KNeighborsClassifier(n_neighbors=k)
stratified_k_fold = StratifiedKFold(n_splits=5)
accuracies_plain = []
accuracies_mpc = []

for train_index, test_index in stratified_k_fold.split(X, y):
    X_train, X_test = X[train_index], X[test_index]
    y_train, y_test = y[train_index], y[test_index]
    
    # Plain
    knn.fit(X_train, y_train)
    y_pred = knn.predict(X_test)
    accuracy = accuracy_score(y_test, y_pred)
    accuracies_plain.append(accuracy)
    
    # Crypten
    cipher_databases = crypten.cryptensor(X_train)    
    y_pred = []
    for i in X_test:
        cipher_query = crypten.cryptensor(i)
        pred = encrypted_knn_search(cipher_databases,cipher_query)
        idx = (pred == 1).nonzero().squeeze().item()
        y_pred.append(y_train[idx])
    accuracy = accuracy_score(y_test, y_pred)
    accuracies_mpc.append(accuracy)

mean_accuracy_plain = np.mean(accuracies_plain)
print('Accuracy for each fold: ', accuracies_plain)
print('Mean accuracy Plain: ', mean_accuracy_plain)
mean_accuracy_mpc = np.mean(accuracies_mpc)
print('Accuracy for each fold: ', accuracies_mpc)
print('Mean accuracy MPC: ', mean_accuracy_mpc)`
```

If i convert numpy array to tensor before passing cryptetensor(), accuracy improved.

from sklearn.datasets import load_iris,load_breast_cancer,load_digits
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import StratifiedKFold
from sklearn.metrics import accuracy_score
import numpy as np

data = load_iris()
X = data.data
y = data.target
print(X.shape)

k = 1
knn = KNeighborsClassifier(n_neighbors=k)
stratified_k_fold = StratifiedKFold(n_splits=5)
accuracies_plain = []
accuracies_mpc = []

for train_index, test_index in stratified_k_fold.split(X, y):
    X_train, X_test = X[train_index], X[test_index]
    y_train, y_test = y[train_index], y[test_index]
    
    # Plain
    knn.fit(X_train, y_train)
    y_pred = knn.predict(X_test)
    accuracy = accuracy_score(y_test, y_pred)
    accuracies_plain.append(accuracy)
    
    # Crypten
    cipher_databases = crypten.cryptensor(torch.tensor(X_train))    
    y_pred = []
    for i in X_test:
        cipher_query = crypten.cryptensor(torch.tensor(i))
        pred = encrypted_knn_search(cipher_databases,cipher_query)
        idx = (pred == 1).nonzero().squeeze().item()
        y_pred.append(y_train[idx])
    accuracy = accuracy_score(y_test, y_pred)
    accuracies_mpc.append(accuracy)

mean_accuracy_plain = np.mean(accuracies_plain)
print('Accuracy for each fold: ', accuracies_plain)
print('Mean accuracy Plain: ', mean_accuracy_plain)
mean_accuracy_mpc = np.mean(accuracies_mpc)
print('Accuracy for each fold: ', accuracies_mpc)
print('Mean accuracy MPC: ', mean_accuracy_mpc)
